diesel/src/pg/metadata_lookup.rs

```diff
@@ -47,7 +47,21 @@ where
                     .store_type(cache_key, type_metadata);
                 PgTypeMetadata(Ok(type_metadata))
             }
-            Err(_e) => PgTypeMetadata(Err(FailedToLookupTypeError::new(cache_key.into_owned()))),
+            Err(_e) => {
+                let metadata_cache = self.get_metadata_cache();
+                let lower_schema = schema.map(|schema| schema.to_lowercase()).unwrap();
+                if let Some(_) = metadata_cache.lookup_type(&PgMetadataCacheKey {
```

Comment:
`metadata_cache.lookup_type(…)` looks up existing cached metadata, without touching the database. That's not what we want in this case. We want to make an explicit query here, by calling the standalone `lookup_type` function from line 64. (Just noticed that both have the same name, which can be irritating)

Additionally the CI suggestion from below is correct in terms of simplifying the code. As the "other" `lookup_type` function returns an `Result<>` the corresponding code looks like:

```suggestion
                if lookup_type(&PgMetadataCacheKey { … }, self).is_ok()
```